plugins/woocommerce/src/Internal/StockNotifications/SettingsController.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\StockNotifications;
+
+/**
+ * Settings controller.
+ */
+class SettingsController extends \WC_Settings_Page {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		// Add a 'Customer stock notifications' section to Products settings.
+		add_filter( 'woocommerce_get_sections_products', array( $this, 'add_customer_stock_notifications_section' ), 100, 1 );
+
+		// Add the Customer Stock Notifications settings.
+		add_filter( 'woocommerce_get_settings_products', array( $this, 'add_customer_stock_notifications_settings' ), 100, 2 );
+
+		// Display admin notices about incompatible settings combinations.
+		add_action( 'admin_notices', array( $this, 'output_admin_notices' ) );
+	}
+
+	/**
+	 * Add a 'Customer stock notifications' section to Products settings.
+	 *
+	 * @param array $sections Products settings sections.
+	 * @return array New Products settings sections.
+	 */
+	public function add_customer_stock_notifications_section( array $sections ): array {
+
+		$section_title = __( 'Customer stock notifications', 'woocommerce' );
+
+		// Add 'Customer stock notifications' section to the Products tab, after Inventory.
+		$inventory_index = array_search( 'inventory', array_keys( $sections ), true );
+		if ( false !== $inventory_index ) {
+			$sections = array_slice( $sections, 0, $inventory_index + 1, true ) +
+				array( 'customer_stock_notifications' => $section_title ) +
+				array_slice( $sections, $inventory_index + 1, null, true );
+		} else {
+			$sections['customer_stock_notifications'] = $section_title;
+		}
+
+		return $sections;
+	}
+
+	/**
+	 * Add the Customer Stock Notifications settings.
+	 *
+	 * @param array  $settings Original settings.
+	 * @param string $section_id Settings section identifier.
+	 * @return array New settings.
+	 */
+	public function add_customer_stock_notifications_settings( array $settings, string $section_id ): array {
+		if ( 'customer_stock_notifications' !== $section_id ) {
+			return $settings;
+		}
+
+		$is_new_shop = \WC_Install::is_new_install();
```

Comment:
I'm not sure this works as expected. How about setting the "allow signups" option to "no" by default here and modifying the option during Woo's installation [here](https://github.com/woocommerce/woocommerce/blob/0a25b2f633d57f71d4dfd1f86922b2430e32dad5/plugins/woocommerce/includes/class-wc-install.php#L1049-L1054)?

Reply:
Done via: https://github.com/woocommerce/woocommerce/pull/58202/commits/a949b0013c6d0e7fdc3dab0efae868fbd2930ed2